Lib/test/test_xml_etree.py

```diff
@@ -2192,12 +2192,6 @@ def test_issue6233(self):
                 b"<?xml version='1.0' encoding='ascii'?>\n"
                 b'<body>t&#227;g</body>')
 
-    def test_issue3151(self):
-        e = ET.XML('<prefix:localname xmlns:prefix="${stuff}"/>')
-        self.assertEqual(e.tag, '{${stuff}}localname')
-        t = ET.ElementTree(e)
-        self.assertEqual(ET.tostring(e), b'<ns0:localname xmlns:ns0="${stuff}" />')
-
```

Comment:
I understand the other changes. For this one, can you explain why this needs to be removed?

Reply:
Hi @ambv, have you checked the commit message at https://github.com/python/cpython/pull/31453/commits/dd7da01325ca32796e139507a38da08886f8f972 ?

Reply:
Indeed I have not. I have now. This will sadly be a breaking change. Was the elevated strictness here security-related as well?

Reply:
> This will sadly be a breaking change.

Could you elaborate? I should note that xmlns has also been about URIs.

> Was the elevated strictness here security-related as well?

Yes, please see https://github.com/libexpat/libexpat/pull/561/files#diff-d1bcab18f24ba66b34aeb2e156f7fde58ef3de1a165514b0fccf0d04c26838f8R3758-R3767 . This allowed code execution through Expat in another application.

Reply:
> Could you elaborate?

While the use was incorrect per spec, clearly parsing what seems to be XML template files was a use case that existed in the wild when BPO-3151 was filed. The curly-brace and dollar sign suggest some ZOPE-related template (or JBOSS, or JavaScript, or the Sun Java System Web Server, etc. etc.). Whatever this usage was, it will now break with expat 2.4.5+

But since this is security-related, there's nothing we can do other than move on.

Reply:
> But since this is security-related, there's nothing we can do other than move on.

@ambv I notice now that (while Expat doesn't do full validation), moving the namespace separator in ElementTree off current `}` could make this work longer. A space or a newline would be other options, for instance.

Reply:
@ambv it's here: https://github.com/python/cpython/blob/2cae93832f46b245847bdc252456ddf7742ef45e/Modules/_elementtree.c#L3660